modules/gateway/peers.go

```diff
@@ -39,6 +40,18 @@ type peer struct {
 	sess muxado.Session
 }
 
+// sessionHeader is sent after the initial version exchange.
+// It prevents peers on different blockchains from connecting to each other,
+// and prevents the gateway from connecting to itself.
+// The receiving peer can set WantConn to false to refuse the connection,
+// and the initiating peer can set WantConn to false
+// if they merely want to confirm that a node is online.
```

Comment:
should be noted that most text editors have some way to auto-wrap content to 80 characters. In vim, you can do this by highlighting the content in visual mode and then typing `gq`